lib/index.js

```diff
@@ -469,13 +500,33 @@ NodeHttpXHR.prototype.send = function (data) {
     this._resp = resp;
     this._responseText = '';
 
-    resp.setEncoding('utf8');
+    // var contentType = resp.headers['content-type'];
+    // TODO: adjust responseType from content-type header if applicable
```

Comment:
yeah, i've been thinking about this. not sure actually how browser impl behave exactly anymore. for example if i put arraybuffer and request a text file i should not get responseText i.e it shouldn't get decoded to characters, and we should get the plain buffer.

but this will already work here in fact like this. maybe looking at content-type isn't even necessary at all in the end.

one case might only be: when we request something binary and expect text. this can be looked at in response-type, when it is not starting with `text/...`. 

also, one could check that when requesting `json` type, the content-type should actually be `application/json`.

will open an issue where this question can be figured out.

Reply:
Sounds good - the use case for examining `content-type` is not immediately obvious to me either.